Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import DesignSystem
+
+struct IPFSCustomGatewayView: View {
+  
+  private static let ipfsTestPath = "ipfs"
+  private static let ipfsTestCID = "bafkqae2xmvwgg33nmuqhi3zajfiemuzahiwss"
+  private static let ipfsTestContent = "Welcome to IPFS :-)"
+  
+  private enum SetButtonStatus {
+    case enabled
+    case disabled
+    case loading
+  }
+  
+  private let ipfsAPI: IpfsAPI
+  @State private var url: String = "https://"
+  @State private var setButtonStatus: SetButtonStatus = .disabled
+  @State private var presentWrongGatewayAlert: Bool = false
+  
+  init(ipfsAPI: IpfsAPI) {
+    self.ipfsAPI = ipfsAPI
+  }
+  
+  var body: some View {
+    List {
+      Section(footer: Text(Strings.Wallet.nftGatewayLongDescription)) {
+        TextField(ipfsAPI.nftIpfsGateway?.absoluteString ?? "", text: $url)
```

Comment:
I think this should unwrap to `"https://"` (the default text when cleared)?

Reply:
fixed in [fa59bc5](https://github.com/brave/brave-ios/pull/6880/commits/fa59bc5e8fdf66b6faffc6228672cfa2b695c87a)